libs/components/packages/src/schematics/utility/visit-project-files.ts

```diff
@@ -0,0 +1,22 @@
+import { Tree } from '@angular-devkit/schematics';
+import { FileVisitor } from '@angular-devkit/schematics/src/tree/interface';
+
+const rootIgnore = ['dist', 'coverage'];
+const alwaysIgnore = ['node_modules'];
```

Comment:
What's the difference between "root" ignore and "always" ignore?

Reply:
It's possible to use `dist` or `coverage` as directories in a project (it'd be unfortunate but not prohibited), so only ignore them at the root of a workspace, whereas `node_modules` should be ignored at any level.